src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -245,11 +245,13 @@ protected virtual void Initialize()
                     // Disable core storage 
                     ScriptConfig.HostConfig.StorageConnectionString = null;
                 }
-
-                // Create the lease manager that will keep handle the primary host blob lease acquisition and renewal 
-                // and subscribe for change notifications.
-                _blobLeaseManager = BlobLeaseManager.Create(storageString, TimeSpan.FromSeconds(15), ScriptConfig.HostConfig.HostId, TraceWriter);
-                _blobLeaseManager.HasLeaseChanged += BlobLeaseManagerHasLeaseChanged;
+                else if (ScriptConfig.RoleDetectionEnabled)
+                {
```

Comment:
This change also ensures we're only creating a `BlobLeaseManager` if a connection string is defined.